### PR TITLE
Add Wavpack (wv) to the list of "Music files".

### PR DESCRIPTION
**Pull request by Yetangitu**
Add "wv" (Wavpack) to the list of "Music files", To actually get Airsonic to play these files to the web player the "wv" format will need to be added the "mp3 audio" transcoder.

### CODE REVIEW

**Comment by randomnicode** [comment]
> To actually get Airsonic to play these files to the web player the "wv" format will need to be added the "mp3 audio" transcoder.

Can we add that?

**Comment by Yetangitu** [comment]
It can be added, yes. It can be added to the default transcoder, adding it to existing installations where admins may have edited the applicable formats takes more effort.

**Comment by randomnicode** [comment]
I'd add it to just the default, since this change is only addressing defaults (maybe use a where value is original string)?

**Comment by Yetangitu** [comment]
Or do this:

```sql
update transcoding set source_formats=((select source_formats from transcoding where name='mp3 audio') || ' wv') where name='mp3 audio';
```

With this precondition:

```sql
select count(*) from transcoding where name='mp3 audio' and (source_formats like '%wv %' or source_formats like '% wv' or source_formats like '% wv %');
```

i,e,:

```xml
<databaseChangeLog
        xmlns="http://www.liquibase.org/xml/ns/dbchangelog"
        xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
        xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-3.1.xsd">
    <changeSet id="add-wavpack" author="yetangitu">
        <preConditions onFail="MARK_RAN">
            <sqlCheck expectedResult="0">select count(*) from transcoding where name='mp3 audio' and (source_formats like 'wv %' or source_formats like '% wv %' or source_formats like '% wv')</sqlCheck>
        </preConditions>
        <sql>update transcoding set source_formats=((select source_formats from transcoding where name='mp3 audio') || ' wv') where name='mp3 audio';</sql>
        <rollback />
    </changeSet>
</databaseChangeLog>
```
This works in all situations and gives consistent results for new installations which will have the `source_formats` set to `ogg oga aac m4a flac wav wma aif aiff ape mpc shn wv`.

**Comment by randomnicode** [comment]
Yup, this is fine.

Two changes I'd make:
- Instead of a sql statement, I'd make it into a liquibase update xml node
- the concat operator || isn't global, you should use concat() instead

**Comment by randomnicode** [comment]
there's a conflict here

**Comment by Yetangitu** [comment]
Sigh - not any more

File: airsonic-main/src/main/resources/liquibase/11.0/add-wavpack.xml [hunk 1]
```diff
@@ -0,0 +1,15 @@
+<databaseChangeLog
+        xmlns="http://www.liquibase.org/xml/ns/dbchangelog"
+        xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+        xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-3.1.xsd">
+    <changeSet id="add-wavpack" author="yetangitu">
+        <preConditions onFail="MARK_RAN">
+            <sqlCheck expectedResult="0">select count(*) from transcoding where name='mp3 audio' and (source_formats like 'wv %' or source_formats like '% wv %' or source_formats like '% wv')</sqlCheck>
+        </preConditions>
+        <update tableName="transcoding">
+            <column name="source_formats" valueComputed="(concat((select source_formats from transcoding where name='mp3 audio'), ' wv'))" />
```
Comment (randomnicode) [inline airsonic-main/src/main/resources/liquibase/11.0/add-wavpack.xml:10]: don't think you need all this.

You can probably just suffice with:
`valueComputed="(concat(source_formats, ' wv'))`

the where clause will take care of the selection

Comment (Yetangitu) [reply airsonic-main/src/main/resources/liquibase/11.0/add-wavpack.xml:10]: True, I just cut and pasted the line from the raw SQL (which I much prefer).

**Pull request merged**